### ISSUE TITLE
False positive for unexpected-keyword-arg using `**` operator

### ISSUE THREAD

**Issue by quimey** [PyCQA/pylint#7293]
### Bug description

```python
def func(arg):
    print(arg)


d = {'b': 'hello world'}
d['arg'] = d['b']
del d['b']

func(**d)
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
test.py:1:0: C0116: Missing function or method docstring (missing-function-docstring)
test.py:9:0: E1123: Unexpected keyword argument 'b' in function call (unexpected-keyword-arg)
```

### Expected behavior

No `unexpected-keyword-arg` as the key 'b' has been deleted from the dictionary

### Pylint version

```shell
pylint @ git+https://github.com/PyCQA/pylint@00dba439171cff81ff68c9302d40279bf916476c
astroid-2.12.2
Python 3.8.0 (default, Dec  9 2021, 17:53:27)
```

### OS / Environment

Ubuntu bionic
